src/channel-qvm.lisp

```diff
@@ -0,0 +1,374 @@
+  (let ((nrs (loop :for rule1 in (noise-rules nm1) 
+                   :append (loop :for rule2 in (noise-rules nm2)
+                                 :collect (make-noise-rule 
+                                           (predicate-and (noise-predicate rule1) (noise-predicate rule2)) 
+                                           (list (operation-elements rule1) (operation-elements rule2)))
+                                 :collect (make-noise-rule 
+                                           (predicate-and (noise-predicate rule1) (predicate-not (noise-predicate rule2))) 
+                                           (operation-elements rule1))
+                                 :collect (make-noise-rule 
+                                           (predicate-and (predicate-not (noise-predicate rule1)) (noise-predicate rule1)) 
+                                           (operation-elements rule2))))))
+    (make-noise-model nrs)))
+
+
+;; The following are examples of predicates! 
+(defun match-strict-qubits (qubits)
+  "The returned function is true if the instruciton's qubits are exactly equal to qubits"
+  (lambda (i-qubits i-name) (equal qubits i-qubits)))
+
+
+(defun match-any-qubits (qubits)
+  "The returned function is true if there is any intersection between the instruction's qubits and qubits."
+  (lambda (i-qubits i-name) (intersection qubits i-qubits )) )
+
+
+(defun match-strict-gates (gates)
+  "The returned function is true if the instruciton's gates are exactly equal to gates"
+  (lambda (i-qubits i-name) (equal gates i-name)))
+
+
+(defun match-any-gates (gates)
+  "The returned function is true if there is any intersection between the instruction's gates and gates."
+  (lambda (i-qubits i-name) (instr-intersect gates i-name )))
+
+
+(defun match-all-gates ()
+  "The returned function is true if the instruction contains a gate (not a measure)."
+  (lambda (i-qubits i-name) (not (equal (list "MEASURE") i-name))))
+
+
+(defun match-measure ()
+  "The returned function is true if the instruction is a MEASURE."
+  (lambda (i-qubits i-name) (equal (list "MEASURE") i-name)))
+
+
+(defun match-measure-at-strict (qubit)
+  "The returned function is true if the instruciton is a measure on the specified qubit."
+  (lambda (i-qubits i-name) (and (equal qubit i-qubits) (string= "MEASURE" i-name))))
+
+
+(defun match-measure-at-any (qubits)
+  "The returned function is true if the instruciton is a measure on any of the specified qubits. "
+  (lambda (i-qubits i-name) (and (intersection qubits i-qubits) (string= "MEASURE" i-name))))
+
+
+(defmethod instr-intersect (l1 l2)
+  "Check if there are any gates in common between l1 and l2. 1 and l2 are  lists of string."
+  (loop named outer :for inst1 in l1
+        do (loop named inner :for inst2 in l2
+                 do (when (string= inst1 inst2) (return-from outer "A")))))
```

Comment:
Would `(intersection l1 l2 :test #'string=)` work for you here instead of these nested `loop`s?

Reply:
yup! I didn't know about using `:test`, and intersection wasn't working because it wasn't testing for string equality. Thanks!